src/pycsou/_dev/fw_utils.py

```diff
@@ -0,0 +1,122 @@
+import typing as typ
+
+import numpy as np
+
+import pycsou.abc.operator as pyco
+import pycsou.runtime as pycrt
+import pycsou.util as pycu
+import pycsou.util.ptype as pyct
+
+
+class SquaredL2Norm(pyco.ProxDiffFunc):
+    # f: \bR^{M} -> \bR
+    #      x     -> \norm{x}{2}^{2}
+    def __init__(self, shape=None):
+        super().__init__(shape=(1, None))
+        self._lipschitz = np.inf
+        self._diff_lipschitz = 2
+
+    @pycrt.enforce_precision(i="arr")
+    def apply(self, arr):
+        xp = pycu.get_array_module(arr)
+        y = xp.linalg.norm(arr, axis=-1, keepdims=True)
+        y2 = xp.power(y, 2, dtype=arr.dtype)
+        return y2
+
+    @pycrt.enforce_precision(i="arr")
+    def grad(self, arr):
+        return 2 * arr
+
+    @pycrt.enforce_precision(i=["arr", "tau"])
+    def prox(self, arr, tau):
+        y = arr / (2 * tau + 1)
+        return y
+
+    @pycrt.enforce_precision(i="data")
```

Comment:
I think you need to add `allow_None=True` in order to be able to handle a `None` input.
`@pycrt.enforce_precision(i="data", allow_None=True)`